src/lib.rs

```diff
@@ -272,21 +243,65 @@ where
     }
 }
 
+fn infer_geom_dimension<T: WktNum + FromStr + Default>(
+    tokens: &mut PeekableTokens<T>,
+) -> Result<Dimension, &'static str> {
+    if let Some(Ok(c)) = tokens.peek() {
+        match c {
+            // If we match a word check if it's Z/M/ZM and consume the token from the stream
+            Token::Word(x) => match x.as_str() {
+                "Z" => {
```

Comment:
These should be case insensitive - see our matching elsewhere using `eq_ignore_ascii_case`